bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php

```diff
@@ -365,23 +365,24 @@ public function getIdPathPagingInfoAction(Request $request)
      */
     public function getAction(Request $request, EventDispatcherInterface $eventDispatcher)
     {
-        // check for lock
-        if (Element\Editlock::isLocked($request->get('id'), 'object')) {
-            return $this->getEditLockResponse($request->get('id'), 'object');
-        }
-        Element\Editlock::lock($request->get('id'), 'object');
-
-        $objectFromDatabase = DataObject::getById(intval($request->get('id')));
+        $objectFromDatabase = DataObject::getById((int)$request->get('id'));
         if ($objectFromDatabase === null) {
             return $this->adminJson(['success' => false, 'message' => 'element_not_found'], JsonResponse::HTTP_NOT_FOUND);
         }
         $objectFromDatabase = clone $objectFromDatabase;
 
         // set the latest available version for editmode
         $latestObject = $this->getLatestVersion($objectFromDatabase);
+        $object = $latestObject;
+
+        // check for lock
+        if (Element\Editlock::isLocked($request->get('id'), 'object') && ($object->isAllowed('save') || $object->isAllowed('publish') || $object->isAllowed('unpublish') || $object->isAllowed('delete'))) {
+            return $this->getEditLockResponse($request->get('id'), 'object');
+        }
+        Element\Editlock::lock($request->get('id'), 'object');
```

Comment:
To be consistent, locking should only be done when user has the rights to modify the object.